packages/cli/src/commands/export.ts

```diff
@@ -19,20 +71,217 @@ export class ExportCommand extends BaseCommand {
+      const emptyFilter = this[FilterFlag.OmitEmpty] ? origin !== EvaluatedConfigOrigin.Empty : true;
+
+      const isKeyPicked = pickedKeys.includes(key);
+      const isLabelPicked = _.intersection(pickedLabels, cfgu.label ?? []).length > 0;
+      const pickFilter = _.isEmpty([...pickedKeys, ...pickedLabels]) || isKeyPicked || isLabelPicked;
+
+      const isKeyOmitted = omittedKeys.includes(key);
+      const isLabelOmitted = _.intersection(omittedLabels, cfgu.label ?? []).length > 0;
+      const omitFilter = !isKeyOmitted && !isLabelOmitted;
+
+      return hiddenFilter && emptyFilter && pickFilter && omitFilter;
+    };
+  }
+
+  async exportConfigs(result: ExportCommandOutput, label: string) {
+    // TODO: needs the parsed result from the export command
+    // if (this.template) {
+    //   const templateContent = await readFile(this.template);
+    //   let templateContext: TemplateContext = configs;
+    //   if (this['template-input'] === 'array') {
+    //     templateContext = _(configs)
+    //       .entries()
+    //       .map(([key, value]) => ({
+    //         key,
+    //         value,
+    //       }))
+    //       .value();
+    //   }
+    //   const { value: compiledContent } = Expression.parse(templateContent).tryEvaluate(templateContext);
+    //   process.stdout.write(compiledContent);
+    //   return;
+    // }
+
+    // TODO: needs to use the Dotenv formatter from the registry
+    // if (this.source) {
+    //   const formattedConfigs = formatConfigs({ format: 'Dotenv', json: configs, label, wrap: true });
+    //   process.stdout.write(formattedConfigs);
+    //   return;
+    // }
+
+    if (this.run) {
+      this.context.configu.runScript(this.run, {
+        cwd: cwd(),
+        // TODO: needs the parsed result from the export command
+        // env: { ...configs, ...process.env },
+      });
+      return;
+    }
+
+    process.stdout.write(result);
+  }
 
   async execute() {
     await this.init();
 
-    const pipe = await this.readPreviousEvalCommandOutput();
-    if (!pipe) {
+    const previousEvalCommandOutput = await this.readPreviousEvalCommandOutput();
+    if (!previousEvalCommandOutput) {
       this.context.stdio.warn('no configuration was fetched');
       return;
     }
 
-    const exportCommand = new BaseExportCommand({ pipe });
-    const { result } = await exportCommand.run();
+    if (this.explain) {
+      this.explainConfigs(previousEvalCommandOutput);
+      return;
+    }
 
+    const label = this.label ?? `configs-${Date.now()}`;
+    const filter = this.filterFromFlags();
+    const keys = this.keysMutations();
+    const pipe = keys
+      ? _.mapValues(previousEvalCommandOutput, (config, key) => ({ ...config, key: keys(key) }))
+      : previousEvalCommandOutput;
+    // TODO: how do we implement the format flag? we somehow have to connect the format flag with the formatter from the registry and pass it via the 'reduce' prop of ExportCommand
```

Comment:
@davidmeirlevy, I need your input here. When providing the `format` flag, we need to fetch the formatter from the registry or download it remotely somehow but it currently doesn't expose expressions or any other logic to do this. How do you think we should implement this logic?